.azure-pipelines/steps/linux.yml

```diff
@@ -0,0 +1,34 @@
+steps:
+- checkout: self
+  fetchDepth: 2
+
+- template: show-environment-variables.yml
+- template: show-disk-usage.yml
+
+- bash: |
+    sudo apt install gdb
+
+    curl -fo $HOME/stamp https://s3-us-west-1.amazonaws.com/rust-lang-ci2/rust-ci-mirror/2017-03-17-stamp-x86_64-unknown-linux-musl
+    chmod +x $HOME/stamp
+
+    export PATH=$PATH:$HOME/.local/bin:$HOME/Library/Python/2.7/bin/:$HOME
+    echo "##vso[task.setvariable variable=PATH;]$PATH"
+
+    mkdir -p $HOME/rustsrc
+  displayName:  Prep
+
+# FIXME: using the job display name feels brittle. Is there a better variable that represents the current matrix key?
+- bash: |
```

Comment:
tl;dr comment is out-of-date and safe to remove

Ah yes I was previously using System.JobDisplayName because it used to be the name of the matrix key, but then the title format changed.  I've asked @vtbassmatt and @willsmythe and there's currently not a variable that is the matrix key.

Reply:
Matrix keys can be defined and used like so:
```
matrix: {
        Win32Debug:   {Arch: 'Win32', Config: 'Debug'},
        Win32Release: {Arch: 'Win32', Config: 'Release'}
}
```
`$(Arch)` and `$(Config)` will both resolve as expected further down the pipeline. It looks like that's the intended use of `$IMAGE` here, referencing a definition in .azure-pipelines/auto.yml